qcengine/exceptions.py

```diff
@@ -0,0 +1,63 @@
+import traceback
+
+
+class QCEngineException(Exception):
+    """
+    Base QCEngine exception, should never be called explicitly.
+    """
+    error_type = "base_error"
+    header = "QCEngine Base Error"
+
+    def __init__(self, message: str):
+
+        # Call the base class constructor with the parameters it needs
+        super().__init__(message)
+
+        # Now for your custom code...
+        self.raw_message = message
+        self.traceback = traceback.format_exc()
+
+    @property
+    def error_message(self) -> str:
+        return f"{self.header}: {self.raw_message}"
+
+
+class UnknownError(QCEngineException):
+    """
+    Unknown QCEngine error, the type was not able to be specified.
+    """
+    error_type = "unknown_error"
+    header = "QCEngine Unknown Error"
+
+
+class InputError(QCEngineException):
+    """
+    Incorrect user parameters, not recoverable. Also may indicate a version issue.
```

Comment:
For this in the bottom comment we have two sets of version errors here:
 1) The version is valid for QCEngine, but a particular method or parameter is only available in a higher version number. This would result in an InputError and be very hard for us to distinguish between a normal input and version issue.
2) The version is too old for QCEngine which throws a ResourceError error as below.

Reply:
I agree with that separation. Especially when I think of multiple clusters where the software may be heterogeneous, rather than the normal single-cluster of homogeneous software and heterogeneous hardware.